src/Uno.UI/UI/Xaml/Controls/Canvas/Canvas.Layout.cs

```diff
@@ -23,16 +23,21 @@ public partial class Canvas : ICustomClippingElement
 	{
 		protected override Size MeasureOverride(Size availableSize)
 		{
+			double maxWidth = 0, maxHeight = 0;
+
 			MeasureOverridePartial();
-			// A canvas does not have dimensions and will always return zero even with a chidren collection.
 			foreach (var child in Children)
 			{
 				if (child is _View)
 				{
-					MeasureElement(child, new Size(double.PositiveInfinity, double.PositiveInfinity));
+					var measuredSize = MeasureElement(child, new Size(double.PositiveInfinity, double.PositiveInfinity));
+
+					maxHeight = Math.Max(maxHeight, measuredSize.Height + GetTop(child as DependencyObject));
+					maxWidth = Math.Max(maxWidth, measuredSize.Width + GetLeft(child as DependencyObject));
 				}
 			}
-			return new Size(0, 0);
+			// It will not be arranged if always return zero.
+			return new Size(maxWidth, maxHeight);
```

Comment:
I just checked the source code of UWP/WinUI for the `<Canvas>` control (something I can't share here for obvious reasons) and the results on Windows is a constant `0, 0`.

So this fix is not appropriate and could break existing applications. Another fix should be found for #5115.